R/ggs_volcano.R

```diff
@@ -0,0 +1,181 @@
+#'
+#' @return A plot returned by the [ggplot()] function
+#' @export
+#'
+#' @examples
+#' library("macrophage")
+#' library("DESeq2")
+#' library("org.Hs.eg.db")
+#' library("AnnotationDbi")
+#'
+#' # dds object
+#' data("gse", package = "macrophage")
+#' dds_macrophage <- DESeqDataSet(gse, design = ~line + condition)
+#' rownames(dds_macrophage) <- substr(rownames(dds_macrophage), 1, 15)
+#' dds_macrophage <- estimateSizeFactors(dds_macrophage)
+#'
+#'
+#' # annotation object
+#' anno_df <- data.frame(
+#'   gene_id = rownames(dds_macrophage),
+#'   gene_name = mapIds(org.Hs.eg.db,
+#'                      keys = rownames(dds_macrophage),
+#'                      column = "SYMBOL",
+#'                      keytype = "ENSEMBL"),
+#'   stringsAsFactors = FALSE,
+#'   row.names = rownames(dds_macrophage)
+#' )
+#'
+#' # res object
+#' data(res_de_macrophage, package = "GeneTonic")
+#' res_de <- res_macrophage_IFNg_vs_naive
+#'
+#' # res_enrich object
+#' data(res_enrich_macrophage, package = "GeneTonic")
+#' res_enrich <- shake_topGOtableResult(topgoDE_macrophage_IFNg_vs_naive)
+#' res_enrich <- get_aggrscores(res_enrich, res_de, anno_df)
+#'
+#' ggs_volcano(res_de,
+#'            res_enrich,
+#'            anno_df,
+#'            geneset_id = res_enrich$gs_id[1]
+#'            )
+#'            
+#' alternatively
+#'
+#' # genelist <- sapply(anno_df$gene_name, function(x) str_contains(x, "CD"))
+#' 
+#' # ggs_volcano(res_de,
+#' #             res_enrich,
+#' #             anno_df,
+#' #             genelist = genelist
+#' #             )
+#' 
+ggs_volcano <- function(res_de,
+                       res_enrich,
+                       annotation_obj = NULL,
+                       gtl = NULL,
+                       geneset_id = NULL,
+                       genelist = NULL,
+                       FDR = 0.05,
+                       color = "red",
+                       volcano_labels = 25,
+                       plot_title = NULL
+                       ) {
+
+  if (!is.null(gtl)) {
+    checkup_gtl(gtl)
+    dds <- gtl$dds
+    res_de <- gtl$res_de
+    res_enrich <- gtl$res_enrich
+    annotation_obj <- gtl$annotation_obj
+  }
+
+  # Retrieve information about genes in geneset gs_id
+  if (!is.null(geneset_id)) {
+    if (geneset_id %in% res_enrich[["gs_id"]]) {
+      thisset_name <- res_enrich[geneset_id, "gs_description"]
+      thisset_members <- unlist(strsplit(res_enrich[geneset_id, "gs_genes"], ","))
+      thisset_members_ids <- annotation_obj$gene_id[match(thisset_members, annotation_obj$gene_name)]
+    }
+  }else {
+    # overwritable via a list
+    if (!all(genelist %in% rownames(res_de))) {
+      not_there <- genelist[!(genelist %in% rownames(res_de))]
+      warning("Some of the provided gene ids were not found in the SummarizedExperiment",
+              "\nNot found: ", not_there)
+    }
+    thisset_members_ids <- intersect(rownames(res_de), genelist)
+    thisset_name <- "Custom list"
+  }
+
+  thisset_members_ids
```

Comment:
I think this one can safely be removed - I also saw it in the signature heatmap function, it is probably a remnant of debugging sessions